extensions/security/iprange/src/org/exist/security/realm/iprange/IPRangeServlet.java

```diff
@@ -0,0 +1,121 @@
+/*
+ *  eXist Open Source Native XML Database
+ *  Copyright (C) 2009-2011 The eXist Project
+ *  http://exist-db.org
+ *  
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public License
+ *  as published by the Free Software Foundation; either version 2
+ *  of the License, or (at your option) any later version.
+ *  
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *  
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ *  
+ *  $Id$
+ */
+package org.exist.security.realm.iprange;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.exist.security.AbstractRealm;
+import org.exist.security.AuthenticationException;
+import org.exist.security.SecurityManager;
+import org.exist.security.Subject;
+import org.exist.xquery.XQueryContext;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+/**
+ * IPRange authenticator servlet.
+ *
+ * @author <a href="mailto:wshager@gmail.com">Wouter Hager</a>
+ */
+public class IPRangeServlet extends HttpServlet {
+
+    protected final static Logger LOG = LogManager.getLogger(IPRangeServlet.class);
+    private static final long serialVersionUID = -568037449837549034L;
+    public static AbstractRealm realm = null;
+
+    public IPRangeServlet() throws ServletException {
+    }
+
+    @Override
+    public void init(final ServletConfig config) throws ServletException {
+        super.init(config);
+    }
+
+    @Override
+    protected void doGet(final HttpServletRequest req, final HttpServletResponse resp)
+            throws ServletException, IOException {
+        doPost(req, resp);
+    }
+
+    @Override
+    protected void doPost(final HttpServletRequest request, final HttpServletResponse response)
+            throws ServletException, IOException {
+
+
+        // Get reverse proxy header when available, otherwise use regular IP address
+        String ipAddress = request.getHeader("X-Forwarded-For");
+        if (ipAddress == null) {
+            ipAddress = request.getRemoteAddr();
+        }
+
+
+        LOG.info("Detected IPaddress " + ipAddress);
+
+        String jsonResponse = "{\"fail\":\"IP range not authenticated\"}";
+
+        try {
+            final SecurityManager securityManager = IPRangeRealm.instance.getSecurityManager();
```

Comment:
This is not nice. At least consider `IPRangeRealm.getInstance()`. Otherwise go through the `BrokerPool` to get the `SecurityManager`